.github/workflows/cronJob.yaml

```diff
@@ -11,29 +11,63 @@ jobs:
     runs-on: ubuntu-latest
     outputs:
       pr_details: ${{ steps.extract.outputs.pr_details }}
+      is_empty: ${{ steps.extract.outputs.is_empty }}
     env:
       API_URL: https://api.github.com/repos/redhat-developer/lsp4ij/pulls
+    name: PR Details
     steps:
       - name: Extract PR numbers and merge_commit_shas
         shell: bash
         id: extract
         run: |
-          pr_infos=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
-          "${{ env.API_URL }}")
-
+          # Fetch PR details from the GitHub API
+          pr_infos=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "${{ env.API_URL }}")
+          
           # Extract PR numbers and merge_commit_sha values, excluding draft pull requests
-          pr_details=$(echo "$pr_infos" | jq -r '.[] | select(.draft == false) | {number: .number, sha: .merge_commit_sha}')
-
-          # Print the PR number and corresponding merge commit sha
-          echo "PR number and merge commit sha:"
-          echo "$pr_details" | jq -r '. | "PR #\(.number): \(.sha)"'
-
-          # Create a JSON array string of PR numbers and SHAs
-          pr_details=$(echo "$pr_details" | jq -nc '[inputs | {number: .number, sha: .sha}]')
-          echo "pr_details=$pr_details" >> $GITHUB_OUTPUT
+          pr_numbers=$(echo "$pr_infos" | jq -r '.[] | select(.draft == false) | {number: .number, sha: .merge_commit_sha} | @base64')          
+          
+          # Array to store PRs that are not drafts and have no merge conflicts
+          declare -a valid_prs=()
+          
+          for pr in $pr_numbers; do
+            # Decode the base64 encoded JSON string
+            pr=$(echo "$pr" | base64 --decode)
+          
+            # Extract PR number
+            number=$(echo "$pr" | jq -r '.number')
+          
+            url="${{ env.API_URL }}/$number"
+            pr_detail=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "$url")
+          
+            mergeable_state=$(jq -r '.mergeable_state' <<< "$pr_detail")
+          
+            if [[ "$mergeable_state" != "dirty" ]]; then
+            pr_number=$(jq -r '.number' <<< "$pr_detail")
+            pr_sha=$(jq -r '.merge_commit_sha' <<< "$pr_detail")
+            valid_prs+=("{\"number\": \"$pr_number\", \"sha\": \"$pr_sha\"}")
+          else
+              echo "::warning file=::PR #$(jq -r '.number' <<< "$pr_detail") has conflicts."
```

Comment:
I'm curious what this looks like in the build results. Would be great if you had a screenshot.

Reply:
@mrglavas 
<img width="1501" alt="image" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/141303561/6149613c-095f-4541-bf39-144dc726ce62">
Currently all lsp4ij PRs have merge conflicts. So it doesnt run tests against the PR.